### ISSUE TITLE
Autocompletion for joined table?

### ISSUE THREAD

**Issue by d33tah** [dbcli/pgcli#393]
Here's what I entered to pgcli:
```select count(distinct col) from ohhai oh join omnomnom om on oh.id=om.```

After the dot, pgcli suggested "om" and "oh", but not "omnomnom" columns. Why is that?

**Comment by amjith** [comment]
It should suggest columns from omnomnom as well. I just verified it on my machine. 

Can you give me some details? 

Such as, version of pgcli, version of postgres and maybe the output of `\d omnomnom`?

**Comment by d33tah** [comment]
```
dbname> \d omnomnom
+---------------+----------+-------------+
| Column        | Type     | Modifiers   |
|---------------+----------+-------------|
| omnm_entry_id | integer  |             |
| om_id         | integer  |             |
| detail        | smallint |             |
+---------------+----------+-------------+
Foreign-key constraints:
    "omnomnom_om_id_fkey" FOREIGN KEY (om_id) REFERENCES omnm_foreign(id)
    "omnomnom_omnm_entry_id_fkey" FOREIGN KEY (omnm_entry_id) REFERENCES omnm_entry(id)

Command Time: 0.000s
Format Time: 0.002s
```

**Comment by d33tah** [comment]
```
dbname> select version();
+---------------------------------------------------------------------------------------------------------------+
| version                                                                                                       |
|---------------------------------------------------------------------------------------------------------------|
| PostgreSQL 9.3.5 on x86_64-unknown-linux-gnu, compiled by gcc (GCC) 4.8.2 20140120 (Red Hat 4.8.2-16), 64-bit |
+---------------------------------------------------------------------------------------------------------------+
SELECT 1
Command Time: 0.000s
Format Time: 0.002s
```

**Comment by d33tah** [comment]
And the pgcli version is the latest in PIP.

**Comment by d33tah** [comment]
Also tested on latest `master`. As before, it only suggests `om` and `oh`.

**Comment by amjith** [comment]
I'm surprised that you only get `om` and `oh` after the dot. 

Here's a screen cap of me trying something very similar and I get the columns from each alias after the dot. 

![join_sample](https://cloud.githubusercontent.com/assets/49260/10671704/fe48e8b6-78a0-11e5-9cb2-9aff2b5babf6.gif)

**Comment by d33tah** [comment]
Two differences: you're using more whitespace and you're using `JOIN aaaa a ON a.col1 = b.col1` while I'm using `JOIN aaaa a ON b.col1 = a.col1`. Perhaps it's one of those?

**Comment by darikg** [comment]
Confirmed it's because of the lack of whitespace. Investigating now.

**Comment by darikg** [comment]
@amjith the issue is `parseutils.last_word(text_before_cursor, include='many_punctuations')`, which doesn't break on the equals sign. It's easy to add the equals sign to the regex, but really all operators e.g. `<>` and `!~` should be in there as well.

**Comment by d33tah** [comment]
@amjith by the way, which program do you use for recording the screen and converting it to gif? You might find https://asciinema.org/ a cleaner solution.

**Comment by amjith** [comment]
I'm using http://www.cockos.com/licecap/ to record my gif. 

I've heard of asciinema, but I haven't used it since I can't embed it into github or something similar.

**Comment by jamiely** [comment]
Related to this, it seems like pgcli doesn't autosuggest after the second table you join, meaning if you join t1, t2, and t3, you do not get suggestions for t3. Should this be the case? If not, I can provide a test case.

**Comment by darikg** [comment]
@jamiely you mean if you don't put whitespace around the join operator? Or more generally? A test case would be appreciated

**Comment by jamiely** [comment]
@darikg Here's a video of me attempting it https://dl.dropboxusercontent.com/u/4073777/pgcli-issue33.mov.

Test case:

Postgres version: 9.4.5
pgcli: 0.20.1

1. Create a new database
2. Run the following creations:
```sql

CREATE TABLE t1 (
  id int PRIMARY KEY,
  name varchar(10) );

CREATE TABLE t2 (
  id int PRIMARY KEY,
  name varchar(10),
  t1_id int);

CREATE TABLE t3 (
  id int PRIMARY KEY,
  name varchar(10),
  t2_id int);
```
3. Now attempt to enter the following query in pgcli:
```sql
select * from t1
          inner join t2 ON
           t1.id = t2.t1_id
          inner join t3 ON
           t2.id = t3.t2_id;
```
Note that autosuggest appears when writing the join statements for t1 and t2, but when we get to t3, autosuggest doesn't suggest "t2_id" nor any field for t3.

**Comment by darikg** [comment]
I have fixes for the original issue and also the bug that @jamiely reported incoming in the next couple days.

**Comment by jamiely** [comment]
Awesome, thanks @darikg!

**Comment by amjith** [comment]
This is now available in v1.0. Please upgrade. 

```
pip install -U pgcli
```
